qualtran/bloqs/arithmetic/comparison_test.py

```diff
@@ -230,7 +230,9 @@ def test_greater_than_manual():
     qlt_testing.assert_wire_symbols_match_expected(
         GreaterThanConstant(bitsize, 17), ['In(x)', '⨁(x > 17)']
     )
-    assert cbloq.t_complexity() == t_complexity(LessThanEqual(bitsize, bitsize))
+    assert cbloq.t_complexity() == (
+        t_complexity(LessThanEqual(bitsize, bitsize)) + TComplexity(clifford=1)
+    )
 
```

Comment:
Can we add a correctness test for the circuit here / tensor contraction / classical simulation?

Reply:
added some classical simulation tests. indeed caught a lot of bugs 😅. now works on random lists of size 2-16

Reply:
Great, (make sure the tests don't take too long or else @mpharrigan will be on your case :))

Reply:
hahaha yep, total is ~0.5s, should be good